homeassistant/components/smappee_official/switch.py

```diff
@@ -0,0 +1,198 @@
+                    dev.append(
+                        SmappeeActuator(
+                            smappee_base,
+                            service_location,
+                            actuator.name,
+                            actuator_id,
+                            actuator.type,
+                            actuator.serialnumber,
+                            actuator_state_option=option,
+                        )
+                    )
+
+    async_add_entities(dev, True)
+
+
+class SmappeeActuator(SwitchEntity):
+    """Representation of a Smappee Comport Plug."""
+
+    def __init__(
+        self,
+        smappee_base,
+        service_location,
+        name,
+        actuator_id,
+        actuator_type,
+        actuator_serialnumber,
+        actuator_state_option=None,
+    ):
+        """Initialize a new Smappee Comfort Plug."""
+        self._smappee_base = smappee_base
+        self._service_location = service_location
+        self._actuator_name = name
+        self._actuator_id = actuator_id
+        self._actuator_type = actuator_type
+        self._actuator_serialnumber = actuator_serialnumber
+        self._actuator_state_option = actuator_state_option
+        self._state = self._service_location.actuators.get(actuator_id).state
+        self._connection_state = self._service_location.actuators.get(
+            actuator_id
+        ).connection_state
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        if self._actuator_type == "INFINITY_OUTPUT_MODULE":
+            return (
+                f"{self._service_location.service_location_name} - "
+                f"Output module - {self._actuator_name} - {self._actuator_state_option}"
+            )
+
+        # Switch or comfort plug
+        return (
+            f"{self._service_location.service_location_name} - "
+            f"{self._actuator_type.title()} - {self._actuator_name}"
+        )
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        if self._actuator_type == "INFINITY_OUTPUT_MODULE":
+            return (
+                self._service_location.actuators.get(self._actuator_id).state
+                == self._actuator_state_option
+            )
+
+        # Switch or comfort plug
+        return self._state == "ON_ON"
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return ICON
+
+    def turn_on(self, **kwargs):
+        """Turn on Comport Plug."""
+        if self._actuator_type in ["SWITCH", "COMFORT_PLUG"]:
+            self._service_location.set_actuator_state(self._actuator_id, state="ON_ON")
+        elif self._actuator_type == "INFINITY_OUTPUT_MODULE":
+            self._service_location.set_actuator_state(
+                self._actuator_id, state=self._actuator_state_option
+            )
+        self._state = self._service_location.actuators.get(self._actuator_id).state
```

Comment:
Polling entities call `async_update` at the end of the service call. So we only need to update the state here if the update feedback takes longer than 2 seconds which is the timeout in the frontend.